Source/Core/Core/IOS/Network/SSL.cpp

```diff
@@ -35,6 +35,40 @@ static constexpr mbedtls_x509_crt_profile mbedtls_x509_crt_profile_wii = {
     0,         /* No RSA min key size */
 };
 
+namespace
+{
+// Dirty workaround to disable SNI which isn't supported by the Wii.
+//
+// This SSL extension can ONLY be disabled by undefining
+// MBEDTLS_SSL_SERVER_NAME_INDICATION and recompiling the library. When
+// enabled, it uses the SNI extension which is sent with the Client Hello
+// message.
+//
+// This workaround defers mbedtls_ssl_set_hostname after the Client Hello. It
+// uses the send callback as it's the (only?) hook used at the beginning of
+// each step during the handshake. If the hostname is NULL, SNI isn't sent
+// with the Client Hello message.
+//
+// However, the hostname needs to be set as it is checked against the Common
+// Name field during the certificate verification process.
```

Comment:
I recommend the following wording: 
`// The Wii doesn't support SNI but the only way to disable it is to disable SSL. 
// We do this by undefining MBEDTLS_SSL_SERVER_NAME_INDICATION and recompiling the library `
This is followed by the third and fourth paragraphs unedited.
In my opinion this is a more concise and clear explanation.

Reply:
But it would be wrong, wouldn't it? The point of this PR instead of #9406 is that we *don't* recompile the library with that flag disabled.

Reply:
Oof, shout I delete that comment? I feel embarrased 😣

Reply:
You don't have to. The wording wasn't clear enough, so I'll rewrite it to something better (hopefully). I'm open to suggestions/better phrasing.